date_range/models/date_range_type.py

```diff
@@ -26,3 +26,10 @@ def _default_company(self):
     _sql_constraints = [
         ('date_range_type_uniq', 'unique (name,company_id)',
          'A date range type must be unique per company !')]
+
+    @api.multi
+    def unlink(self):
```

Comment:
Instead of this, use `ondelete='restrict'` in the many2one in date.range.

Reply:
@pedrobaeza I know, but i wanted to track the exact error, and to benefit from it to super method in https://github.com/OCA/account-financial-tools/pull/636, because there the super was not covered in tests, red build, and i only had an postgres integrity error, and wanted to track it as odoo error..

Reply:
You can test it the same. Just use:

```python
with self.assertRaises(Exception):
    with self.env.cr.savepoint():
        self.date_range_type.unlink()
```

Reply:
@pedrobaeza Updated, but i think is fine if i leave also the unlink method, to have a nice error message for the user.

Reply:
But this is code to maintain. Generic error is clear enough IMO.